pandas/tests/frame/indexing/test_indexing.py

```diff
@@ -1074,10 +1103,12 @@ def test_setitem_single_column_mixed_datetime(self):
         tm.assert_series_equal(result, expected)
 
         # GH#16674 iNaT is treated as an integer when given by the user
-        df.loc["b", "timestamp"] = iNaT
-        assert not isna(df.loc["b", "timestamp"])
-        assert df["timestamp"].dtype == np.object_
-        assert df.loc["b", "timestamp"] == iNaT
+        if not using_array_manager:
+            # TODO(ArrayManager) setting iNaT in DatetimeArray actually sets NaT
```

Comment:
one way to avoid this is be using ensure_wrapped_if_datetimelike in ArrayManager.setitem to ensure you have a DTA and never a ndarray[dt64]